endpoint/src/datamodel.rs

```diff
@@ -21,10 +21,19 @@ use uuid::Uuid;
 pub(crate) struct ProductFootprint {
     pub(crate) id: PfId,
     pub(crate) spec_version: SpecVersionString,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) preceding_pf_ids: Option<Vec<PfId>>,
```

Comment:
`preceding_pf_ids` must be either undefined or a non-empty set. I am afraid we must somehow wrap this to inject this property into the resulting OpenAPI Document.